### ISSUE TITLE
Training hangs after scanning training images

### ISSUE THREAD

**Issue by marcmk6** [ultralytics/ultralytics#23034]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

```
AMP: running Automatic Mixed Precision (AMP) checks...
AMP: checks passed ✅
train: Fast image access ✅ (ping: 3.4±5.0 ms, read: 175.8±72.7 MB/s, size: 524.2 KB)
train: Scanning /mnt/data/relevant_frames/labels/train... 590503 images, 79 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 590503/590503 677.1it/s 14:32
```
The program seems to hang after scanning all the images. How to identify the problem? This happens no matter I use one GPU or multiple. 
The data cache seems never created successfully. The program makes a full scan of training data every time. 

I found this https://github.com/ultralytics/ultralytics/issues/8896#issuecomment-2046665069 , but after wrapping the code into a function, the model even doesn't get created...
I'm in a Linux env with NVIDIA GPUs

**Comment by UltralyticsAssistant** [comment]
👋 Hello @marcmk6, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Thanks for the detailed log snippet—since the run appears to stop right after dataset scanning, this could be an environment/dataloader/process-sync situation. This is an automated response 🤖, and an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. In particular, please include:
-Exact command or Python snippet you run (including all args)
-Your `data.yaml` (paths can be anonymized)
-Your full console log from launch to “hang”
-`ultralytics` version, Python version, PyTorch version, CUDA version, GPU model(s)
-Whether this is single-process, DDP, or multiple workers, and the values you set for workers/batch/imgsz/etc.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Also, since you referenced prior guidance, please share the exact “wrapped into a function” code you tried and the resulting output (even if the model doesn’t get created) 🧩.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
That “hang” right after `Scanning ...labels/train...` is often either the large label `*.cache` file being written (can take a while on mounted/NFS volumes) or a dataloader worker deadlock; can you paste the output of `yolo checks` plus the full terminal log after the scan (run once with `workers=0` and `verbose=True`), and also confirm `/mnt/data/relevant_frames/labels/` is writable so Ultralytics can create something like `labels/train.cache` (cache saving is skipped if the cache dir isn’t writable)? See the relevant background in the [Troubleshooting Common YOLO Issues](https://docs.ultralytics.com/guides/yolo-common-issues/) guide.

```bash
yolo checks
yolo train model=yolo11n.pt data=/path/to/data.yaml workers=0 verbose=True
```

**Comment by marcmk6** [comment]
Thanks for your help! It turns out a network storage problem. Moved the data to local storage then it runs smoothly.

**Comment by glenn-jocher** [comment]
That makes sense—training + cache writes can stall badly on high-latency network mounts; sticking to local SSD/scratch for both `images/` and `labels/` (so `*.cache` can be written quickly) is the right fix, and if you ever must train from a mount you can try `cache=disk` to force local `.npy` caching and reduce I/O sensitivity (see [Model training tips](https://docs.ultralytics.com/guides/model-training-tips/)).
